pkg/cqrs/base_cqrs/cqrs.go

```diff
@@ -93,6 +93,13 @@ func (w wrapper) LoadFunction(ctx context.Context, envID, fnID uuid.UUID) (*stat
 		return nil, err
 	}
 
+	app, err := w.GetAppByID(ctx, fn.AppID)
+	if err != nil {
+		return nil, err
+	}
+	// Append the app name as the user facing "app id"
+	def.AppID = app.Name
```

Comment:
Note - this approach would need to be changed in prod to use the cache for app lookups due to the throughput. Even with the cache, adding an additional lookup for every execution would add more load on the cache and potentially a perf hit in execution. May need to re-consider to perhaps only load this upon finalize.